### PR TITLE
Add the start of specification documentation.

### PR DESCRIPTION
**Pull request by bigkevmcd**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        services:
+          - name: app-1-service-http
+            # the "source_url" represents the "upstream" source code e.g. the
+            # Java/Go/Node.js code.
+            #
+            # This is used when generating the EventListener to drive the CI
+            # process, PullRequests and Pushes to this branch drive the
+            # pipelines.
+            source_url: https://github.com/myproject/myservice.git
+          - name: app-1-service-metrics
```
Comment (sbose78) [inline docs/README.md:38]: For those with no `source_url`, it implies that the service is not to be built off a repo, rather all kube manifests would be present in the 'gitops' repo?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      - name: my-app-1
+        services:
+          - name: app-1-service-http
+          - name: app-1-service-metrics
+            # the "config_repo" property is an upstream reference to
+            # configuration in another repository.
+            #
+            # This is used by the ArgoCD creation to drive deployments of
+            # configuration from this repository.
+            config_repo:
```
Comment (sbose78) [inline docs/README.md:59]: Fair enough.

File: docs/README.md (hunk 1 above)
Comment (bigkevmcd) [reply docs/README.md:38]: No, it means that we wouldn't include it in the eventlistener...

Comment (bigkevmcd) [reply docs/README.md:38]: That's not to say that it's not more generally useful for documentation...

**Comment by mnuttall** [comment]
```
# This is a set of "apps", which according to the IBM/Red Hat repository
     # "contain" a set of services.
     apps:
       - name: my-app-1
         services:
           - name: app-1-service-http
```

Service aggregation into applications is currently implemented by code such as https://github.com/rhd-gitops-example/gitops-example-dev/blob/master/apps/app-1/base/kustomization.yaml. This means that `kubectl apply -k apps/app-1` will deploy all the services that comprise that application. 

This PR offers a different way of recording this aggregation. How are they expected to work together? Which is the source of truth? Would changing one cause the other to be updated? 

In general, it would be useful to know what the scope of this tool is. Is it to be an uber-source-of-truth? Is the tool intended to be able to react to any changes to any of its stanzas by effectively making those things so?

**Comment by bigkevmcd** [comment]
@mnuttall good questions, I think so, yes.

In my head there there's an "--overwrite" option to rewrite files that are deemed to be defined by the manifest.

But, given that this is intended to be "generate the changes for a PR" it might be reasonable to assume that they can be overwritten.

**Pull request closed without merging**